src/Configuration/src/KubernetesBase/KubernetesConfigMapProvider.cs

```diff
@@ -121,11 +134,52 @@ private void ProcessData(V1ConfigMap item)
             {
                 foreach (var data in item?.Data)
                 {
-                    configMapContents[NormalizeKey(data.Key)] = data.Value;
+                    // Should this check for application.{env.Environment}.json as well?
+                    if (string.Equals(data.Key, ConfigFileKey, StringComparison.InvariantCultureIgnoreCase))
+                    {
+                        using var stream = this.GenerateStreamFromString(data.Value);
+                        var jsonConfiguration = ParseConfigMapFile(stream);
+
+                        // Give precedence to overrides in the configmap, so ignore duplicate keys
+                        foreach (var jsonKey in jsonConfiguration.Keys.Where(k => !item.Data.Keys.Contains(k)))
```

Comment:
Is this necessary? I would think that if you want to use appsettings.json in a configmap that that is the only key you'd use in that one and then override it with a separate configmap... I could be wrong but that's my initial thought

Reply:
I considered this necessary to make determining which settings more easily expressed. With this approach, it can just be documented that values in the configmap will override any values in the appsettings.json config value. Without this, it will be dependent on the order key/values are processed in the configmap.

Reply:
But this code executes within the context of a single configmap ~ likely at least 2 instances of the `KubernetesConfigMapProvider` have been created ([appname and appname.envName](https://github.com/SteeltoeOSS/Steeltoe/blob/main/src/Configuration/src/KubernetesBase/KubernetesConfigurationBuilderExtensions.cs#L45)). It seems odd to me that anybody would want to combine standard key/value pairs and one or more appsettings.json style sets of configuration within a single configmap.

Overall this isn't a huge issue and isn't a great deal of complexity whether its needed or not, but it goes against the ways I would envision configmaps being used. If this class expects either standard key/value pairs or json provided under a single key (and eventually yaml or props or whatever else) then we don't need to worry about the last-entry-wins type of logic already provided by IConfiguration

Reply:
I had overlooked 2 instances of the `KubernetesConfigMapProvider` being created. I can make that update.

Reply:
Resolved in [97f89f3e3e64ce33b778f90b1f8a54585c334ce1](https://github.com/SteeltoeOSS/Steeltoe/pull/671/commits/97f89f3e3e64ce33b778f90b1f8a54585c334ce1)